src/main/java/com/google/cloud/tools/maven/DeployAllMojo.java

```diff
@@ -46,6 +46,15 @@ public void execute() throws MojoExecutionException, MojoFailureException {
       if (yaml.exists()) {
         getLog().info("deployAll: Preparing to deploy " + yamlName);
         deployables.add(yaml);
+      } else if (!isStandardStaging()) {
```

Comment:
I was thinking this logic could probably be organized as flexible vs. standard as the top-level condition, rather than having the standard yaml existence be the default check. In the case that the user was using flexible but had yamls in the standard locations, then it would be deploying possibly incorrect yamls. It might be fine, but it seems a bit off to me.

Reply:
As far as I understand, in flexible projects we do want to check for yamls to deploy in both the source directory and the staging directory, but I'm not sure. @loosebazooka?

Reply:
Oh wait I see what you're saying I think